src/main/java/it/gov/pagopa/payhub/auth/mypay/service/MyPayOperatorsService.java

```diff
@@ -0,0 +1,38 @@
+package it.gov.pagopa.payhub.auth.mypay.service;
+
+import it.gov.pagopa.payhub.auth.mypay.model.MyPayOperator;
+import it.gov.pagopa.payhub.auth.mypay.repository.MyPayOperatorsRepository;
+import java.util.Optional;
+import java.util.Set;
+import org.springframework.stereotype.Service;
+
+@Service
+public class MyPayOperatorsService {
+
+  private final MyPayOperatorsRepository myPayOperatorsRepository;
+
+  public MyPayOperatorsService(MyPayOperatorsRepository myPayOperatorsRepository){
+    this.myPayOperatorsRepository = myPayOperatorsRepository;
+  }
+
+  public void registerMyPayOperator(String mappedExternalUserId,String email, String organizationIpaCode,
+      Set<String> roles) {
+
+    Optional<MyPayOperator> existingMyPayOperator = Optional.ofNullable(
```

Comment:
you could modify the repository method to return directly the Optional instead of wrapping explicitly you